sql/driver/wire.go

```diff
@@ -80,14 +100,17 @@ func (d Datum) Value() (driver.Value, error) {
 		val = t.BytesVal
 	case *Datum_StringVal:
 		val = t.StringVal
+	case *Datum_DateVal:
+		val = Date{t.DateVal.GoTime()}
```

Comment:
The client and server each have a time zone to be aware of; this only works when they are the same. I believe we will need #2327 to have a good time zone policy; until then the only consistent policy is to use UTC everywhere. Introducing new dependencies on the server's system timezone is a step in the wrong direction.